File: 정종길/프로그래머스/카카오 1회독/2022 KAKAO BLIND RECRUITMENT/k진수에서 소수 개수 구하기 .java
import java.util.*;

class Solution {
    public int solution(int n, int k) {
        int answer = 0;

        String kBinaryNum = kBinary(n, k);
        String[] kBinaryNums = kBinaryNum.split("0");

        for(String num : kBinaryNums) {
            if(!num.isEmpty()){
                if(checkPrimeNumber(Long.parseLong(num))){
                    answer += 1;
                }
            }
        }

        return answer;
    }

    public String kBinary(int n, int k){
        String kNum = "";
        while(n != 0){
            kNum = (n % k) + kNum;
            n /= k;
        }
        return kNum;
    }

    public boolean checkPrimeNumber(long num) {
        if (num == 1){
            return false;
        }

        long sqrt = (long) Math.sqrt(num);
        for(int i = 2 ; i <= sqrt; i++) {
            if(num % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: 정종길/프로그래머스/카카오 2회독/2022 KAKAO BLIND RECRUITMENT/k진수에서 소수 개수 구하기.py
def solution(n, k):
    answer = 0
    nums = convert(n, k).replace("0", " ").split()
    
    for num in nums:
        if check_prime(int(num)):
            answer += 1
            
    return answer

def convert(n, k):
    q, r = divmod(n, k)
    if q == 0:
        return str(r)
    return convert(q, k) + str(r)

def check_prime(num):
    if num <= 1:
        return False
    
    if num == 2:
        return True
    
    if num % 2 == 0:
        return False
    
    for i in range(3, int(num ** 0.5 + 1), 2):
        if num % int(i) == 0:
            return False
    return True